controllers/operator_controller.go

```diff
@@ -162,14 +162,21 @@ func (r *OperatorReconciler) reconcile(ctx context.Context, op *operatorsv1alpha
 		return ctrl.Result{}, err
 	}
 
-	// update operator status
-	apimeta.SetStatusCondition(&op.Status.Conditions, metav1.Condition{
-		Type:               operatorsv1alpha1.TypeReady,
-		Status:             metav1.ConditionTrue,
-		Reason:             operatorsv1alpha1.ReasonResolutionSucceeded,
-		Message:            "resolution was successful",
-		ObservedGeneration: op.GetGeneration(),
-	})
+	// convert existing unstructured object into bundleDeployment for easier mapping of status.
+	existingTypedBundleDeployment := &rukpakv1alpha1.BundleDeployment{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(dep.UnstructuredContent(), existingTypedBundleDeployment); err != nil {
+		apimeta.SetStatusCondition(&op.Status.Conditions, metav1.Condition{
+			Type:               operatorsv1alpha1.TypeReady,
+			Status:             metav1.ConditionUnknown,
```

Comment:
Leaving it to `Unknown` instead of `False`, since an error in converting to typed object does not necessarily mean that install was unsuccessful.